### PR TITLE
update merkeldag to use the explicit decoder registry

### PR DESCRIPTION
**Pull request by willscott**
Update the interface of `go-ipld-legacy` to reflect the change in https://github.com/ipfs/go-ipld-legacy/pull/12

There is no breaking interface change in this update.

will wait until v0.2.0 of legacy is tagged and then use that version as the dependency here.

needs the equivalent update in the boxo fork of this repo i think.

### CODE REVIEW

**Review by rvagg — APPROVED** [review]
very nicely nonbreaking
👍 when the version is bumped and go mod is tidied

**Comment by aschmahmann** [comment]
Up to you. Might be easier in case we find any other oddities along the way. Sorry I'm the slow leg here.

However, if you don't want things to exactly match anymore that's ok too. e.g. if boxo adds a setter for the registry and you don't want to.

**Comment by willscott** [comment]
ending up with 2 variants of this code with slightly different semantics so that it's harder to migrate seems worse than keeping them in sync

**Comment by willscott** [comment]
@aschmahmann  - this should now include #105  and matches what's in boxo

File: merkledag.go [hunk 1]
```diff
@@ -143,22 +145,33 @@ func (sg *sesGetter) Get(ctx context.Context, c cid.Cid) (format.Node, error) {
...
-	return getNodesFromBG(ctx, sg.bs, keys)
+	return getNodesFromBG(ctx, sg.bs, keys, sg.decoder)
 }
 
 // WrapSession wraps a blockservice session to satisfy the format.NodeGetter interface
 func WrapSession(s *bserv.Session) format.NodeGetter {
-	return &sesGetter{s}
+	d := legacy.NewDecoder()
+	d.RegisterCodec(cid.DagProtobuf, dagpb.Type.PBNode, ProtoNodeConverter)
+	d.RegisterCodec(cid.Raw, basicnode.Prototype.Bytes, RawNodeConverter)
```
Comment (aschmahmann) [inline merkledag.go:160]: this could just be `ipldLegacyDecoder` instead of remaking the same structure.

**Pull request merged**